pkg/maintainer/spv/redemptions.go

```diff
@@ -0,0 +1,312 @@
+			logger.Infof(
+				"skipped proving redemption transactions for wallet [%x] "+
+					"because of wallet state [%v]",
+				walletPublicKeyHash,
+				wallet.State,
+			)
+			continue
+		}
+
+		walletTransactions, err := btcChain.GetTransactionsForPublicKeyHash(
+			walletPublicKeyHash,
+			transactionLimit,
+		)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"failed to get transactions for wallet: [%v]",
+				err,
+			)
+		}
+
+		for _, transaction := range walletTransactions {
+			isUnproven, err :=
+				isUnprovenRedemptionTransaction(
+					transaction,
+					walletPublicKeyHash,
+					btcChain,
+					spvChain,
+				)
+			if err != nil {
+				return nil, fmt.Errorf(
+					"failed to check if transaction is an unproven redemption "+
+						"transaction: [%v]",
+					err,
+				)
+			}
+
+			if isUnproven {
+				unprovenRedemptionTransactions = append(
+					unprovenRedemptionTransactions,
+					transaction,
+				)
+			}
+		}
+	}
+
+	return unprovenRedemptionTransactions, nil
+}
+
+func isUnprovenRedemptionTransaction(
+	transaction *bitcoin.Transaction,
+	walletPublicKeyHash [20]byte,
+	btcChain bitcoin.Chain,
+	spvChain Chain,
+) (bool, error) {
+	// If the transaction does not have exactly one input, it cannot be a
+	// redemption transaction.
+	if len(transaction.Inputs) != 1 {
+		return false, nil
+	}
+
+	singleInput := transaction.Inputs[0]
+
+	// Check whether the single input is the current wallet main UTXO.
+	isMainUtxo, err := isInputCurrentWalletsMainUTXO(
+		singleInput.Outpoint.TransactionHash,
+		singleInput.Outpoint.OutputIndex,
+		walletPublicKeyHash,
+		btcChain,
+		spvChain,
+	)
+	if err != nil {
+		return false, fmt.Errorf(
+			"failed to check if input is the main UTXO",
+		)
+	}
+
+	// If the single input is not the current main UTXO of the wallet, the
+	// transaction is either a redemption transaction that is already
+	// proven or it's not a redemption transaction at all.
+	if !isMainUtxo {
+		return false, nil
+	}
+
+	changeFound := false
+
+	// Look at the transaction's outputs. All the outputs must be pending
+	// redemption requests, except for one optional change output.
+	for _, output := range transaction.Outputs {
+		// First, check if the given output is a change (if it wasn't
+		// found yet).
+		if !changeFound {
+			isChange, err := isWalletChangeOutput(walletPublicKeyHash, output)
+			if err != nil {
+				return false, fmt.Errorf(
+					"failed to check if output is wallet change: [%v]",
+					err,
+				)
+			}
+
+			if isChange {
+				changeFound = true
+				continue
+			}
+		}
+
+		// If the given output is not a change, it must be a pending redemption
+		// request.
+		_, err := spvChain.GetPendingRedemptionRequest(
+			walletPublicKeyHash,
+			output.PublicKeyScript,
+		)
+		if err != nil {
+			if errors.Is(err, tbtc.ErrPendingRedemptionRequestNotFound) {
+				// This output is neither a change nor a pending request.
+				// That means this is not a redemption transaction.
+				return false, nil
```

Comment:
Wouldn't it be better to refactor the `GetPendingRedemptionRequest` in the similar manner as we did with GetDepositRequest:
```
	// GetDepositRequest gets the on-chain deposit request for the given
	// funding transaction hash and output index.The returned values represent:
	// - deposit request which is non-nil only when the deposit request was
	//   found,
	// - boolean value which is true if the deposit request was found, false
	//   otherwise,
	// - error which is non-nil only when the function execution failed. It will
	//   be nil if the deposit request was not found, but the function execution
	//   succeeded.
	GetDepositRequest(
		fundingTxHash bitcoin.Hash,
		fundingOutputIndex uint32,
	) (*tbtc.DepositChainRequest, bool, error)
```
If we added the boolean value indicating whether or not the pending redemption was found, we wouldn't have to
rely on error comparison. The fact that the pending redemption was not found is not really an error.